modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableIndexStoragesSupplier.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.table.distributed;
+
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Supplier table index storages.
+ */
+public interface TableIndexStoragesSupplier {
+    /**
+     * Returns indexes by their ID.
+     *
+     * <p>Waits for the primary key index and all other registered indexes to be created.
+     */
+    Map<UUID, TableSchemaAwareIndexStorage> get();
+
+    /**
+     * Adds a wait to create an index, if not already created, on a subsequent call to {@link #get()}.
```

Comment:
This comment is hard to understand, please rephrase it

Reply:
Try fix it